built_in_decorators.py

```diff
@@ -0,0 +1,219 @@
+from abc import ABC, abstractmethod
+from functools import singledispatch
+
+
+class Movable(ABC):
+    """
+    Class that extends "ABS" built-in class.
+    """
+    def __init__(self):
+        """
+        Constructor
+        """
+        pass
+
+    @abstractmethod
+    def move(self):
+        """
+        Abstract method that need to be implement in the successors of this class.
+        """
+        pass
+
+
+class Player(Movable):
+    """
+    Class that extends "Movable" class.
+    """
+    def __init__(self):
+        """
+        Constructor
+        """
+        super().__init__()
+
+    def move(self):
+        """
+        Implementation of the abstract method of its "parent".
+        """
+        print("can move to all direction")
+
+
+class Enemy(Movable):
+    """
+    Class that extends "Movable" class.
+    """
+    def __init__(self):
+        """
+        Constructor
+        """
+        super().__init__()
+
+    def move(self):
+        """
+        Implementation of the abstract method of its "parent".
+        """
+        print("can move only right or left")
+
+
+###################################
+
+@singledispatch
```

Comment:
Its a cool decorator! I didn't know it before. Im pretty sure you are using it wrong - In the way you used it here, you could have just used typing.Union to accept all the different parameters. The operator > (lt) has different implementations depending on the type.  I think the idea of this decorator is to enable different implementations. 
Look at here for the union example - 
https://towardsdatascience.com/simplify-your-functions-with-functools-partial-and-singledispatch-b7071f7543bb